File: ValidMountainArray.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        def check_ascending(arr):
            if(len(arr)<2): return False
            for i in range(1,len(arr)):
                if(arr[i-1]>=arr[i]):
                    return False
            
            return True
        
        idx=arr.index(max(arr))
    
        return True if(check_ascending(arr[:idx+1]) and check_ascending(list(reversed(arr[idx:])))) else False

File: ValidMountainArray.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int ascend=-1; //init -1, ascend 1 decrease 0
        bool stopped=false;
        
        for(int i=1;i<arr.size();i++){
            if(arr[i-1]<arr[i]){
                if(ascend==0){
                    stopped=true;
                    break;
                }
                ascend=1;
            }else if(arr[i-1]>arr[i]){
                if(ascend==-1){
                    stopped=true;
                    break;
                }
                ascend=0;
            }else{ // arr[i-1]=arr[i]
                stopped=true;
                break;
            }
        }
        
        return !stopped && ascend==0;
    }
};
